CMakeLists.txt

```
@@ -41,15 +41,36 @@ FIND_PACKAGE( DD4hep REQUIRED )
 FIND_PACKAGE( ROOT REQUIRED )
 FIND_PACKAGE( MarlinTrk REQUIRED )
 
+OPTION( USE_SDHCALCONTENT "Use SDHCALContent" OFF )
+OPTION( USE_APRILCONTENT "Use APRILContent" OFF )
 
-FOREACH( pkg PandoraSDK Marlin MarlinUtil  LCContent DD4hep ROOT MarlinTrk )
+FOREACH( pkg PandoraSDK Marlin MarlinUtil LCContent DD4hep ROOT MarlinTrk )
     IF( ${pkg}_FOUND )
         INCLUDE_DIRECTORIES( SYSTEM ${${pkg}_INCLUDE_DIRS} )
         LINK_LIBRARIES( ${${pkg}_LIBRARIES} )
         ADD_DEFINITIONS ( ${${pkg}_DEFINITIONS} )
     ENDIF()
 ENDFOREACH()
 
+IF( USE_SDHCALCONTENT )
+    FIND_PACKAGE( SDHCALContent REQUIRED )
+    INCLUDE_DIRECTORIES( SYSTEM ${SDHCALContent_INCLUDE_DIRS} )
+    LINK_LIBRARIES( ${SDHCALContent_LIBRARIES} )
+    ADD_DEFINITIONS( "-DSDHCALCONTENT" )
+ENDIF()
+
+IF( USE_APRILCONTENT )
+    SET(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${CMAKE_CURRENT_SOURCE_DIR}/cmake)
+    FIND_PACKAGE( APRILContent REQUIRED )
+    FIND_PACKAGE(PkgConfig REQUIRED)
+    PKG_CHECK_MODULES(MLPACK REQUIRED mlpack)
```

```suggestion
    FIND_PACKAGE(mlpack REQUIRED)
```
should also work, I think. Newer versions of mlpack should come with cmake support. If this is not working we might need to tweak the `CMAKE_PREFIX_PATH`. (This will only be necessary until mlpack is also part of the stack. Once that is the case we will take care of populating `CMAKE_PREFIX_PATH` properly).

---

Hi @tmadlener 
cf. https://github.com/iLCSoft/DDMarlinPandora/pull/29#discussion_r1848608827

Did they add the cmakeconfig stuff again?

---

I was mainly looking at the [spack package](https://github.com/spack/spack/blob/develop/var/spack/repos/spack_repo/builtin/packages/mlpack/package.py) where it is listed as a `CMakePackage` so I just assumed that it will be possible. I have missed that part of the discussion.

I think this is not the biggest issue at this point and we can resolve this transparently in a follow up, once we have clarified this a bit further.

---

OK. This seems to be the easiest way to get mlpack via cmake. I checked a spack installation of `mlpack` and it does indeed not have cmake support for downstream dependencies.